crates/typst-realize/src/lib.rs

```diff
@@ -559,9 +560,14 @@ fn prepare(
     // materialization, so that it includes the synthesized fields. Do it before
     // marking as prepared so that show-set rules will apply to this element
     // when queried.
-    let tags = elem
-        .location()
-        .map(|loc| (Tag::Start(elem.clone()), Tag::End(loc, key)));
+    let tags = elem.location().map(|loc| {
+        let flags = TagFlags {
+            introspectable: elem.can::<dyn Introspectable>(),
+            tagged: elem.can::<dyn Tagged>(),
```

Comment:
You wanted to save two lookups by adding `Located`, but here you are doing them anyway and now it's three checks 😅